src/sentry/lang/native/appconnect.py

```diff
@@ -0,0 +1,236 @@
+
+    def __init__(
+        self,
+        api_credentials: appstore_connect.AppConnectCredentials,
+        itunes_cookie: str,
+        itunes_org: int,
+        app_id: str,
+    ) -> None:
+        """Internal init, use one of the classmethods instead."""
+        self._api_credentials = api_credentials
+        self._session = requests.Session()
+        self._itunes_cookie = itunes_cookie
+        self._itunes_org = itunes_org
+        self._app_id = app_id
+
+    @classmethod
+    def from_project(cls, project: Project, config_id: str) -> "AppConnectClient":
+        """Creates a new client for the project's appStoreConnect symbol source.
+
+        This will load the configuration from the symbol sources for the project if a symbol
+        source of the ``appStoreConnect`` type can be found which also has matching
+        ``credentials_id``.
+        """
+        config = get_app_store_config(project, config_id)
+        return cls.from_config(config)
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "AppConnectClient":
+        """Creates a new client from an appStoreConnect symbol source config.
+
+        This config is normally stored as a symbol source of type ``appStoreConnect`` i an
+        project's ``sentry:symbol_sources`` property.
+        """
+        try:
+            jsonschema.validate(config, APP_STORE_CONNECT_SCHEMA)
+        except jsonschema.exceptions.ValidationError as e:
+            raise InvalidConfigError from e
+
+        api_credentials = appstore_connect.AppConnectCredentials(
+            key_id=config["appconnectKey"],
+            key=config["appconnectPrivateKey"],
+            issuer_id=config["appconnectIssuer"],
+        )
+        itunes_cookie = config["itunesSession"]
+        itunes_org = int(config["orgId"])
+        app_id = config["appId"]
+        return cls(
+            api_credentials=api_credentials,
+            itunes_cookie=itunes_cookie,
+            itunes_org=itunes_org,
+            app_id=app_id,
+        )
+
+    def itunes_client(self) -> ITunesClient:
+        """Returns an iTunes client capable of downloading dSYMs.
+
+        This will raise an exception if the session cookie is expired.
+        """
+        return ITunesClient(itunes_cookie=self._itunes_cookie, itunes_org=self._itunes_org)
+
+    def list_builds(self, kind: BuildKind = BuildKind.ALL) -> List[BuildInfo]:
+        """Returns the available builds, grouped by release.
+
+        :param kind: Whether to only query pre-releases or only releases or all.
+        :param bundle: The bundle ID, e.g. ``io.sentry.sample.iOS-Swift``.
+        """
+        if kind == BuildKind.PRE_RELEASE:
+            ret = appstore_connect.get_pre_release_version_info(
+                self._session, self._api_credentials, self._app_id
+            )
+            all_results = {"pre_releases": ret}
+        elif kind == BuildKind.RELEASE:
+            ret = appstore_connect.get_pre_release_version_info(
```

Comment:
should this call a different version other than `pre_release`?

Reply:
oops, good catch